test/integration/sonobuoy_integration_test.go

```diff
@@ -113,31 +138,129 @@ func TestUseNamespaceFromManifest(t *testing.T) {
+	defer cleanup()
+
+	args := fmt.Sprintf("run --image-pull-policy IfNotPresent --wait --mode=quick -n %v", ns)
+	mustRunSonobuoyCommandWithContext(ctx, t, args)
+
+	checkStatusForPluginErrors(ctx, t, ns, "e2e", 0)
+	tb := mustDownloadTarball(ctx, t, ns)
+	tb = saveToArtifacts(t, tb)
+
+	checkTarballPluginForErrors(t, tb, "e2e", 0)
+}
+
+func checkStatusForPluginErrors(ctx context.Context, t *testing.T, ns, plugin string, failCount int) {
+	var expectVals []string
+
+	switch {
+	case failCount == 0:
+		expectVals = []string{
+			`"status":"complete","result-status":"passed"`,
+			`"passed":1`,
+		}
+	case failCount > 0:
+		expectVals = []string{
+			`"status":"complete","result-status":"failed"`,
+			fmt.Sprintf(`"passed":%v`, failCount),
+		}
+	case failCount < 0:
+		t.Fatalf("failCount < 0 not permitted; expected >=0, got %v", failCount)
+	}
+
+	args := fmt.Sprintf(`status --json -n %v`, ns)
+	out := mustRunSonobuoyCommandWithContext(ctx, t, args)
+	for _, v := range expectVals {
+		if !strings.Contains(out.String(), v) {
+			t.Errorf("Expected output of %q to contain %q but output was %v", args, v, out.String())
+		}
+	}
+}
+
+func mustDownloadTarball(ctx context.Context, t *testing.T, ns string) string {
+	args := fmt.Sprintf("retrieve -n %v", ns)
+	tarballName := mustRunSonobuoyCommandWithContext(ctx, t, args)
+	t.Logf("Tarball downloaded to: %v", tarballName.String())
+	return strings.TrimSpace(tarballName.String())
+}
+
+// checkPluginForErrors runs multiple checks to ensure that failCount errors occurred for the given
+// plugin. Ensures that all our different reporting methods are in agreement.
+func checkTarballPluginForErrors(t *testing.T, tarball, plugin string, failCount int) {
+	if plugin == "e2e" {
+		expectOut := fmt.Sprintf("failed tests: %v", failCount)
+		args := fmt.Sprintf("e2e %v ", tarball)
+		out := mustRunSonobuoyCommand(t, args)
+		if !strings.Contains(out.String(), expectOut) {
+			t.Errorf("Expected output of %q to contain %q but output was %v", args, expectOut, out.String())
+		}
+	}
+
+	expectOut := fmt.Sprintf("Failed: %v", failCount)
+	args := fmt.Sprintf("results %v --plugin %v", tarball, plugin)
+	out := mustRunSonobuoyCommand(t, args)
+	if !strings.Contains(out.String(), expectOut) {
+		t.Errorf("Expected output of %q to contain %q but output was %v", args, expectOut, out.String())
+	}
+}
+
+func saveToArtifacts(t *testing.T, p string) (newPath string) {
+	artifactsDir := os.Getenv("ARTIFACTS_DIR")
+	if artifactsDir == "" {
+		t.Logf("Skipping saving artifact %v since ARTIFACTS_DIR is unset.", p)
+	}
+
+	artifactFile := filepath.Join(artifactsDir, filepath.Base(p))
+	origFile := filepath.Join(pwd(t), filepath.Base(p))
+
+	if err := os.MkdirAll(artifactsDir, 0755); err != nil {
+		t.Logf("Error creating directory %v: %v", artifactsDir, err)
+		return p
 	}
 
+	var stdout, stderr bytes.Buffer
+
+	cmd := exec.CommandContext(context.Background(), bash, "-c", fmt.Sprintf("mv %v %v", origFile, artifactFile))
```

Comment:
Need to add a comment on here about why we do this instead of `os.Rename`; it wont be clear to readers.